src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -298,7 +336,7 @@ public bool Remove(T item)
 
         #region IEnumerable Members
 
-        IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+        IEnumerator IEnumerable.GetEnumerator() => new Enumerator(this);
```

Comment:
This had been a common idiom for having `IEnumerable.GetEnumerator()` return the same thing as the `GetEnumerator()` that is part of the class's API. Now it requires comparison to know this is the case. Since `GetEnumerator()` is pretty much guaranteed to be inlined into this, nothing is gained.